flax/core/frozen_dict.py

```diff
@@ -156,13 +157,30 @@ def tree_flatten(self) -> Tuple[Tuple[Dict[Any, Any]], Tuple[()]]:
     """
     return (self._dict,), ()
 
+  def tree_flatten_with_keys(self) -> Tuple[Tuple[Tuple[Any, Any]], Tuple[()]]:
+    """Flattens this FrozenDict.
+
+    Returns:
+      A flattened version of this FrozenDict instance.
+    """
+    return ((jax.tree_util.GetAttrKey('dict'), self._dict),), ()
```

Comment:
This is tricky. There should be a way to return no key at all since this is more of a proxy type.

Reply:
Alternatively, even though it would be slower, we could do a single-level flattening e.g:

```python
children = tuple(((jax.tree_util.GetAttrKey(key), value) for key, value in self._dict.items())
return children, ()
```

Reply:
Thank you Cristian! Your comment was addressed :)